src/main/drivers/accgyro/accgyro_spi_icm426xx.c

```diff
@@ -186,65 +206,34 @@ void icm426xxGyroInit(gyroDev_t *gyro)
     spiWriteReg(dev, ICM426XX_RA_PWR_MGMT0, ICM426XX_PWR_MGMT0_TEMP_DISABLE_OFF | ICM426XX_PWR_MGMT0_ACCEL_MODE_LN | ICM426XX_PWR_MGMT0_GYRO_MODE_LN);
     delay(15);
 
-    uint8_t odrIdx = 0;
-    bool supportedODRFound = false;
-
-    if (gyro->gyroRateKHz) {
-        const uint8_t gyroSyncDenominator = gyro->mpuDividerDrops + 1; // rebuild it in here, see gyro_sync.c
-        const uint8_t desiredODR = 8 / gyroSyncDenominator;
-        for (unsigned i = 0; i < ARRAYLEN(odrConfigLUT); i++) {
-            if (odrConfigLUT[i].odr == desiredODR) {
-                odrIdx = odrConfigLUT[i].odr_idx;
-                supportedODRFound = true;
-                break;
-            }
-        }
-    }
-
-    if (!supportedODRFound) {
-        odrIdx = 6;
+    // Get desired output data rate
+    uint8_t odrConfig = 0;
+    if (gyro->gyroRateKHz && gyro->mpuDividerDrops < ODR_CONFIG_COUNT) {
+        odrConfig = odrLUT[gyro->mpuDividerDrops];
+    } else {
+        odrConfig = odrLUT[ODR_CONFIG_1K];
         gyro->gyroRateKHz = GYRO_RATE_1_kHz;
     }
```

Comment:
`mpuDividerDrops`  was used as number of skipped samples. So mpuDividerDrops==3 shall map to 2kHz, but it will map to 1kHz with current code ... 

That code is unused, but for completeness, support it (another option is to remove this code completely and always use 8k)
```
#define ICM426XX_GYRO_ODR_8K 3
#define ICM426XX_GYRO_ODR_4K 4
#define ICM426XX_GYRO_ODR_2K 5
#define ICM426XX_GYRO_ODR_1K 6
 
static uint8_t odrLUT[]= {
ICM426XX_GYRO_ODR_8K,          // 0 
ICM426XX_GYRO_ODR_4K, ICM426XX_GYRO_ODR_4K,       // 1,2 
ICM426XX_GYRO_ODR_2K, ICM426XX_GYRO_ODR_2K, ICM426XX_GYRO_ODR_2K, ICM426XX_GYRO_ODR_2K, // 3,4,5,6 - 2k
ICM426XX_GYRO_ODR_1K          // 7 - 1k
};
```

```suggestion
    uint8_t odrConfig;
    if (gyro->gyroRateKHz && gyro->mpuDividerDrops < ARRAYLEN(odrLUT)) {
        odrConfig = odrLUT[gyro->mpuDividerDrops];
    } else {
        odrConfig = ICM426XX_GYRO_ODR_1K ;
        gyro->gyroRateKHz = GYRO_RATE_1_kHz;
    }
```

Reply:
Good catch, I should have noticed (but didn't). Good thing you're reviewing this. I tried a solution where we don't need additional macros.